### DISCUSSION TITLE
Zustand vs useState in microfrontend architecture

### DISCUSSION THREAD

**Discussion by seba-cod3** [pmndrs/zustand#1229] [Q&A] [answered]
Hello, i'm analyzing the best way to manage a global state for an SPA with a microfronts architecture.

While investigating arround internet difference between state management libraries vs creating custom global state managment, i had come to a point to determine that i'll be using react-tracked as state management (i have use it in 5 projects so far and it's great, sorry i dont point this Q&A directly in tracked repo but couldnt find the Discussions tab active there).

Note: Redux/ Tracked with redux is a no go as it's not the easiest library to learn and i need this to have the abilitie to scale with new devs joining the team.

With that beign said.. i have been using `createContainer` with `useState` that works great,. (little workarround with lists of objects updating a single inner object key) but i have a question that could not find a straightforward answer.

Is there any downfall to using `useState` as store state managment that `zustand` is here to solve? Will it make it more robust, or is not really necessary given that proxies envolve the store objects?

I'll be happy to read from you, thanks!

**Comment by dai-shi** [comment]
If you prefer, feel free to open a new issue in react-tracked as Discussion isn't enabled there.

**Comment by dai-shi** [comment, marked as answer]
I'm not quite sure about the question, but the difference between useState and zustand is whether it's internal or external. Roughly speaking, internal state is more React oriented.

**Reply by seba-cod3** [reply]
You are already helping me, i'll use useState, thanks.

P.S. (aclaration of the question): i was trying to figure it out if zustand will give my app robustness.. after a POC i've proved that it wasn't necessary and it was only adding a layer of difficulty to the code.

Have a great day!
